pdns/dnsdist-web.cc

```diff
@@ -100,10 +100,8 @@ static bool compareAuthorization(YaHTTP::Request& req, const string &expected_pa
     /* if this is a request for the API,
        check if the API key is correct */
     if (req.url.path=="/jsonstat" ||
-        req.url.path=="/api/v1/servers/localhost" ||
-        req.url.path=="/api/v1/servers/localhost/config" ||
-        req.url.path=="/api/v1/servers/localhost/config/allow-from" ||
-        req.url.path=="/api/v1/servers/localhost/statistics") {
+        req.url.path == "/api/v1" ||
+        req.url.path.find("/api/v1/") == 0) {
       header = req.headers.find("x-api-key");
```

Comment:
Just checking if the path begins with `/api` (or is `/jsonstat`) might be easier and could prevent the issue resurfacing if we ever do `/api/v2` for example?